pkg/operator/controller/controller.go

```diff
@@ -279,7 +312,7 @@ func (r *reconciler) ensureRouterNamespace() error {
 
 // ensureRouterForIngress ensures all necessary router resources exist for a
 // given clusteringress.
-func (r *reconciler) ensureRouterForIngress(ci *ingressv1alpha1.ClusterIngress, dnsConfig *configv1.DNS, infraConfig *configv1.Infrastructure) error {
+func (r *reconciler) ensureRouterForIngress(ci *ingressv1alpha1.ClusterIngress, result *reconcile.Result) error {
```

Comment:
Mutating input is rarely expected by a function caller absent any docs about it. How about just returning`(Result, error)`?

Reply:
We need the same result object to be mutated otherwise the caller need to inspect intermediate result object and update itself that can be ignored by passing the original result object.

Reply:
> We need the same result object to be mutated otherwise the caller need to inspect intermediate result object and update itself that can be ignored by passing the original result object.

Inspecting the intermediate object shouldn't be necessary, although
```go
			// result was declared earlier.
			var err error
			result, err = r.ensureRouterForIngress(ingress, dnsConfig, infraConfig, result)
			// ensureRouterForIngress may have modified `result` or not.
```